spec/support/shared_entity_specs.rb

```diff
@@ -133,3 +133,58 @@ def verify_signature(pubkey, signature, signed_string)
     end
   end
 end
+
+shared_examples "a JSON Entity" do
+  it "#to_json output matches JSON schema" do
+    json = described_class.new(data).to_json
+    expect(json.to_json).to match_json_schema(:entity_schema)
+  end
+
+  let(:to_json_output) { described_class.new(data).to_json.to_json }
+
+  it "contains described_class property matching the entity class (underscored)" do
+    expect(to_json_output).to include_json(entity_type: described_class.entity_name)
+  end
+
+  it "contains JSON properties for each of the entity properties with the entity_data property" do
```

Comment:
Then I think a test for `#to_json` is better, like we have it for xml: `"produces correct XML"` -> `"produces correct JSON"`, instead of just parsing it without any checks.

> Also I think let(:json) makes it easier to understand the code for newcomers.

👍